File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
       
       # build a has map and have COUNT as key, VALUE as value
       # use bucket sort
        hashMap = {}
        frequency = [[] for i in range(len(nums) + 1)]
        
        for item in nums:
            hashMap[item] = 1 + hashMap.get(item, 0)

        for n,c in hashMap.items():
            frequency[c].append(n)

        result = []

        for i in range(len(frequency) - 1, 0, -1):
            for n in frequency[i]:
                result.append(n)
            if len(result) == k:
                return result
             

File: Data Structures & Algorithms/binary-search/submission-1.swift
class Solution {
    func search(_ nums: [Int], _ target: Int) -> Int {
        var L = 0
        var R = nums.count - 1

        while L <= R {
            let midpoint = (L + R) / 2
            if nums[midpoint] > target {
                R = midpoint - 1
            } else if nums[midpoint] < target {
                L = midpoint + 1
            } else {
                return midpoint
            }
        }

        return -1
    }
}
